eventorganizer.api/src/main/java/com/event/organizer/api/service/EventService.java

```diff
@@ -208,4 +209,83 @@ public void userIsGoingToEvent(String username, Long eventId) throws EventOrgani
 
         eventRepository.save(event);
     }
+
+    public List<Event> getThisWeeksEvents() {
+        List<Event> allEvents = eventRepository.findAll();
+
+        LocalDateTime now = LocalDateTime.now();
+        LocalDateTime startOfTheWeek = now.with(TemporalAdjusters.previousOrSame(DayOfWeek.MONDAY));
+        LocalDateTime endOfTheWeek = startOfTheWeek.plusWeeks(1);
+
+        List<Event> eventsThisWeek = allEvents.stream()
+                .filter(event -> event.getStartDate().isAfter(startOfTheWeek) && event.getEndDate().isBefore(endOfTheWeek))
+                .collect(Collectors.toList());
+
+        return eventsThisWeek;
+    }
+
+    public List<Event> getThisMonthsEvents() {
+        List<Event> allEvents = eventRepository.findAll();
+
+        LocalDateTime now = LocalDateTime.now();
+        LocalDateTime startOfTheMonth = now.with(TemporalAdjusters.firstDayOfMonth());
+        LocalDateTime endOfTheMonth = now.with(TemporalAdjusters.lastDayOfMonth());
+
+        List<Event> eventsThisWeek = allEvents.stream()
+                .filter(event -> event.getStartDate().isAfter(startOfTheMonth) && event.getEndDate().isBefore(endOfTheMonth))
+                .collect(Collectors.toList());
+
+        return eventsThisWeek;
+    }
+
+    public List<Event> getEventsByUserLocation(String userLocation) {
```

Comment:
- This is not necessary. Use `getEventsByLocation` instead
- Just pass the "userLocation" from the controller.